src/query.c

```diff
@@ -562,20 +570,43 @@ static IndexIterator *Query_EvalWildcardQueryNode(QueryEvalCtx *q, QueryNode *qn
   IndexSpec *spec = q->sctx->spec;
   Trie *t = spec->terms;
   ContainsCtx ctx = {.q = q, .opts = &qn->opts};
+  RSToken *token = &qn->verb.tok;
 
-  if (!t || !qn->verb.tok.str) {
+  if (!t || !token->str) {
     return NULL;
   }
 
-  qn->verb.tok.len = Wildcard_RemoveEscape(qn->verb.tok.str, qn->verb.tok.len);
+  token->len = Wildcard_RemoveEscape(token->str, token->len);
   size_t nstr;
-  rune *str = strToFoldedRunes(qn->verb.tok.str, &nstr);
+  rune *str = strToFoldedRunes(token->str, &nstr);
 
   ctx.cap = 8;
   ctx.its = rm_malloc(sizeof(*ctx.its) * ctx.cap);
   ctx.nits = 0;
 
-  TrieNode_IterateWildcard(t->root, str, nstr, rangeIterCb, &ctx, &q->sctx->timeout);
+  // spec support contains queries
+  if (spec->suffix) {
+    // all modifier fields are supported
+    if (qn->opts.fieldMask == RS_FIELDMASK_ALL ||
+       (spec->suffixMask & qn->opts.fieldMask) == qn->opts.fieldMask) {
+      SuffixCtx sufCtx = {
+        .root = spec->suffix->root,
+        .rune = str,
+        .runelen = nstr,
+        .cstr = token->str,
+        .cstrlen = token->len,
+        .type = SUFFIX_TYPE_WILDCARD,
+        .callback = charIterCb, // the difference is weather the function receives char or rune
+        .cbCtx = &ctx,
+        .timeout = &q->sctx->timeout,
+      };
+      Suffix_IterateWildcard(&sufCtx);
+    } else {
+      QueryError_SetErrorFmt(q->status, QUERY_EGENERIC, "Contains query on fields without WITHSUFFIXTRIE support");
```

Comment:
Why we decide to fail here instead of fallback to `TrieNode_IterateWildcard` ?

Reply:
Fix for both Trie and TrieMap

Reply:
But you still set an error here, what am I missing?

Reply:
This has been fixed and in case of failure, falls back to brute force solution.